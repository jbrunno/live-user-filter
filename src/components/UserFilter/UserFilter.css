.search-input {
  width: 240px;
  border: none;
  padding: 8px 25px;
  border-radius: 50px;
  outline: none;
  background: rgba(0, 0, 0, 0.3);
  color: #fff;
  font-size: 14px;
}

.search-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  row-gap: 8px;
}