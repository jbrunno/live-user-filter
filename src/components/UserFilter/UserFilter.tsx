import React from 'react';
import './UserFilter.css';

interface UserFilterProps {
  searchTerm: string;
  onSearchTermChange: (newSearchTerm: string) => void;
}

const UserFilter: React.FC<UserFilterProps> = ({
  searchTerm,
  onSearchTermChange,
}) => {
  return (
    <div className="search-container">
      <label htmlFor="search">Search by name or location</label>
      <input
        className="search-input"
        placeholder="Search..."
        id="search"
        type="text"
        value={searchTerm}
        onChange={(e) => onSearchTermChange(e.target.value)}
      />
    </div>
  );
};

export default UserFilter;
