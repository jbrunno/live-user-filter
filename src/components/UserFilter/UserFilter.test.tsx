import { render, fireEvent } from '@testing-library/react';
import UserFilter from './UserFilter';

test('renders correctly', () => {
  const searchTerm = 'example';
  const onSearchTermChange = jest.fn();

  const { getByLabelText } = render(
    <UserFilter
      searchTerm={searchTerm}
      onSearchTermChange={onSearchTermChange}
    />
  );

  const searchInput = getByLabelText('Search by name or location'); // Atualize para o texto correto

  fireEvent.change(searchInput, { target: { value: 'new value' } });

  expect(onSearchTermChange).toHaveBeenCalledWith('new value');
  expect(searchInput).toMatchSnapshot();
});
