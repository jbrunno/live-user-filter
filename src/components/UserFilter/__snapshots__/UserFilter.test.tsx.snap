// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<input
  class="search-input"
  id="search"
  placeholder="Search..."
  type="text"
  value="example"
/>
`;
